mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxPath.java

```diff
@@ -147,6 +147,12 @@ public MailboxPath sanitize(char delimiter) {
         return this;
     }
 
+    public boolean hasEmptyNameInHierarchy(char pathDelimiter) {
+        return this.name.isEmpty()
+            || this.name.contains("..")
+            || this.name.startsWith(".")
+            || this.name.endsWith(".");
```

Comment:
Shouldn't you check if the name `contains`, `startWith` and `endWith` the `pathDelimiter` that you pass as param? Otherwise your function is hardcoded and your parameter useless?

Reply:
ha, nice catch :)
can you add a test about this?